qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,90 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""Base class for calibration-type experiments."""
+
+from abc import abstractmethod
+from typing import Optional
+
+from qiskit.providers.options import Options
+from qiskit.providers.backend import Backend
+
+from qiskit_experiments.framework.base_experiment import BaseExperiment
+from qiskit_experiments.framework.experiment_data import ExperimentData
+
+
+class BaseCalibrationExperiment(BaseExperiment):
+    """An abstract base class for calibration experiments.
+
+    This abstract base class specifies an experiment and how to update an
+    optional instance of :class:`Calibrations` specified in the experiment options
+    under calibrations. Furthermore, the experiment options also specifies
+    an auto_update variable which, by default, is set to True. If this variable,
+    is True then the run method of the experiment will call :meth:`block_for_results`
+    and update the calibrations instance.
+    """
+
+    # The updater class that updates the Calibrations instance
+    __updater__ = None
```

Comment:
What is the difference of defining a class updater and having multiple protected methods to implement each part of processing flow in this class? IIRC the motivation of implementing the updater classes is the inheritance of core code. I think this aspect can be ported to this class.

Reply:
It makes the code more modular and the update methodology reusable. As you can see the `update_calibrations` methods in the experiments is pretty light-weight. I would leave this aspect as is in this PR. If we see that we start adding too much experiment specific logic in the updaters as we add more experiments then we can think about how to best change this.